test/support/pages/common.js

```diff
@@ -244,34 +246,35 @@ export default (function () {
       .then(function () { self.debug('... sleep(' + sleepMilliseconds + ') end'); });
     },
 
-    handleError: function (testObj) {
-      var self = this;
-      var testName = (testObj.parent) ? [testObj.parent.name, testObj.name].join('_') : testObj.name;
+    handleError(testObj) {
+      const self = this;
+      const testName = (testObj.parent) ? [testObj.parent.name, testObj.name].join('_') : testObj.name;
 
       return function (reason) {
-        var now = Date.now();
-        var filename = ['failure', now, testName].join('_') + '.png';
+        const now = Date.now();
+        const fileName = `failure_${now}_${testName}.png`;
 
-        return self.saveScreenshot(filename)
+        return self.saveScreenshot(fileName, true)
         .finally(function () {
           throw reason;
         });
       };
     },
 
-    saveScreenshot: function saveScreenshot(filename) {
-      var self = this;
-      var outDir = path.resolve('test', 'output');
+    async saveScreenshot(fileName, isFailure = false) {
+      const directoryName = isFailure ? 'failure' : 'session';
+      const directoryPath = path.resolve(`test/screenshots/${directoryName}`);
+      const filePath = path.resolve(directoryPath, fileName);
 
-      return self.remote.takeScreenshot()
-      .then(function writeScreenshot(data) {
-        var filepath = path.resolve(outDir, filename);
-        self.debug('Taking screenshot "' + filepath + '"');
-        fs.writeFileSync(filepath, data);
-      })
-      .catch(function (err) {
-        self.log('SCREENSHOT FAILED: ' + err);
-      });
+      this.debug(`Taking screenshot "${filePath}"`);
+
+      try {
+        const screenshot = await this.remote.takeScreenshot();
+        mkdirp.sync(directoryPath);
+        fs.writeFileSync(filePath, screenshot);
+      } catch (err) {
```

Comment:
Probable move `catch` clause one level up, so we `saveScreenshot.catch(reason => console.log(reason))`. Leave error handling to consumer